src/RA_AchievementSet.cpp

```diff
@@ -421,38 +421,43 @@ BOOL AchievementSet::LoadFromFile(GameID nGameID)
     const std::string sFilename = GetAchievementSetFilename(nGameID);
 
     SetCurrentDirectory(NativeStr(g_sHomeDir).c_str());
-    FILE* pFile = nullptr;
-    errno_t nErr = fopen_s(&pFile, sFilename.c_str(), "r");
-    if (pFile != nullptr)
+
+    std::ios::sync_with_stdio(false);
+
+    if (std::ifstream ifile{ sFilename }; !ifile.is_open())
+    {
+        //	Cannot open file
+        RA_LOG("Cannot open file %s\n", sFilename.c_str());
+        RA_LOG("Error %s\n", ra::GetLastErrorMsg());
+        return FALSE;
+    }
+    else
     {
         //	Store this: we are now assuming this is the correct checksum if we have a file for it
         g_pCurrentGameData->SetGameID(nGameID);
 
         if (m_nSetType == Local)
         {
-            const char EndLine = '\n';
-            char buffer[4096];
-            unsigned long nCharsRead = 0;
+            auto buffer{ std::make_unique<char[]>(4096) };
```

Comment:
If you care about size (which is the only reason I can think of for using C arrays) then prefer this approach.
If you don't just just `std::basic_string `and use `data()` when you want to write to it.

Reply:
This code will have to change to address #82